File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/other/JigsawPieceCurrentPosition.java
package com.kafkaquest.kq.common.util.models.events.other;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class JigsawPieceCurrentPosition {

    private UUID pieceId;
    private int rowPosition;
    private int columnPosition;
    private Boolean isInCorrectPosition;
}

File: kq-projection-dotnet/kq-projection/Common/Models/Events/JigsawPieceCurrentPosition.cs
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Events;

public class JigsawPieceCurrentPosition
{
    [JsonPropertyName("pieceId")]
    public Guid pieceId { get; set; }

    [JsonPropertyName("rowPosition")]
    public int rowPosition { get; set; }

    [JsonPropertyName("columnPosition")]
    public int columnPosition { get; set; }

    [JsonPropertyName("isInCorrectPosition")]
    public bool isInCorrectPosition { get; set; }
}
